[sc-80919] new PAPI - dataset get full info

NawreCode opened the pull request:
new public Python API to retrieve all the information about a dataset.

marchelleboid commented on dataikuapi/dss/dataset.py:
```diff
@@ -523,6 +522,17 @@ def get_metric_history(self, metric, partition=''):
+        Retrieve all the information about a dataset
+
+        Returns:
+            a complex JSON object containing all the information on a dataset, such as params, schema, lastbuild infos, status, etc.
```
```suggestion
            a complex JSON object containing all the information about a dataset, such as params, schema, last build infos, status, etc.
```

thtrunck reviewed (approved):
LGTM

cstenac reviewed (changes requested):
* Let's avoid "old-style" APIs where methods directly return raw dicts, as they hard to work with:
    * Create a dataset DSSDatasetInfo
    * get_full_info -> get_info
    * get_info returns a DSSDatasetInfo
    * DSSDatasetInfo has a .raw property returning the dict
    * Add accessor properties on DSSDatasetInfo, and notably a `last_build` which returns either a timestamp or `datetime.datetime`
* Beware, in Python you never return "JSON objects", but "dicts"
* Use modern :arg: :return:, .. comments

NawreCode commented:
@cstenac I've addressed your comments. Tell me if you see anything else which has to be exposed in the `DSSDatasetInfo` class. I've kept it simple and focused on the last build for now.

cstenac commented on dataikuapi/dss/dataset.py:
```diff
@@ -859,3 +870,78 @@ def already_exists(self):
+    """
+
+    def __init__(self, dataset, info):
+        """
```
Nitpick: we don't normally document constructors that we ask customers not to call

NawreCode replied:
removed 👍🏻

cstenac commented on dataikuapi/dss/dataset.py:
```diff
@@ -859,3 +870,78 @@ def already_exists(self):
+        """
+        return self.info
+
+    def get_raw_last_build(self):
```
I don't think this method is useful

NawreCode replied:
I guess you're right. Removed 👍🏻

cstenac commented on dataikuapi/dss/dataset.py:
```diff
@@ -859,3 +870,78 @@ def already_exists(self):
+        """
+        return self.info.get("lastBuild", None)
+
+    def get_last_build_start_time(self, as_date=False):
```
I would put as_date = True by default.

cstenac replied:
Actually, to be consistent with scenario.py, I would:

* Remove the option as_date.
* Always return a date
* Make it a property

check scenario.py 510 and 517

NawreCode replied:
done 👍🏻

cstenac commented on dataikuapi/dss/dataset.py:
```diff
@@ -859,3 +870,78 @@ def already_exists(self):
+            return datetime.datetime.fromtimestamp(timestamp / 1000)
+        return timestamp
+
+    def is_last_build_successful(self):
```
make it a property. Check what we do in scenario.py (462/466/...)

NawreCode replied:
done 👍🏻

cstenac commented on dataikuapi/dss/dataset.py:
```diff
@@ -523,6 +524,16 @@ def get_metric_history(self, metric, partition=''):
+        """
+        Retrieve all the information about a dataset
+
+        :returns: a :class:`DSSDatasetInfo` containing all the information about a dataset, such as params, schema, last build infos, status, etc.
```
I would probably not mention that it contains "params, schema, status", since we provide no accessors for these.

Beware, "information" is always singular.

NawreCode replied:
removed 👍🏻

cstenac reviewed (changes requested):
Minor feedback, but we can still get it a bit better :)
